pandas/core/indexes/base.py

```diff
@@ -4193,15 +4193,43 @@ def putmask(self, mask, value):
             # coerces to object
             return self.astype(object).putmask(mask, value)
 
-    def equals(self, other) -> bool:
+    def equals(self, other: Any) -> bool:
         """
-        Determine if two Index objects contain the same elements.
+        Determine if two Index objects contain the same elements with the same order.
 
         Returns
         -------
         bool
-            True if "other" is an Index and it has the same elements as calling
-            index; False otherwise.
+            True if "other" is an Index and it has the same elements and order
+            as the calling index; False otherwise.
+
+        Examples
+        --------
+        >>> idx1 = pd.Index([1, 2, 3])
+        >>> idx1
+        Int64Index([1, 2, 3], dtype='int64')
+        >>> idx1.equals(pd.Index([1, 2, 3]))
+        True
+
+        The dtype is been compared as well
```

Comment:
The dtype is actually *not* compared:

```
In [9]: pd.Int64Index([1, 2, 3]).equals(pd.UInt64Index([1, 2, 3]))                                                                                                                            
Out[9]: True
```

(in your example here, the values are not equal as a string is not equal to an integer)